src/lib/operators/export_binary.cpp

```diff
@@ -77,6 +78,12 @@ void export_values(std::ofstream& ofstream, const std::vector<pmr_string>& value
 
 // specialized implementation for bool values
 template <>
+void export_values(std::ofstream& ofstream, const pmr_vector<bool>& values) {
+  // Cast to fixed-size format used in binary file
+  const auto writable_bools = std::vector<BoolAsByteType>(values.begin(), values.end());
+  export_values(ofstream, writable_bools);
```

Comment:
Code duplication with void export_values(std::ofstream& ofstream, const std::vector<bool>& values).

Reply:
Same for `void export_values(std::ofstream& ofstream, const pmr_vector<pmr_string>& values) {` (which is not your fault).

Suggestion: https://gist.github.com/mrks/6d6e9892850d694e396de857840c2c7a

Not sure what your C++ level is - if you want to discuss this approach, feel free to stop by.

Reply:
This just came to my mind: Why do we have `std::vector` here at all? All data (data in value segments, dictionaries, attribute vectors, run lengths, ...) should be stored in `pmr_vector`s. Unless I am missing something, the `std::vector` versions should be thrown out and any segment type that depends on it is wrong.

Reply:
Feel free to mark such issues as resolved once you have addressed them. At least when it's clear that there are no remaining open questions/issues.